### PR TITLE
[Notifier] Add ContextBlock for slack notifier

### PR DESCRIPTION
**Pull request by norkunas**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

This block allows to add small text to the message layout footer

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            throw new \LogicException('Maximum number of elements should not exceed 10.');
+        }
+
+        $element = [
+            'type' => $markdown ? 'mrkdwn' : 'plain_text',
+            'text' => $text,
+        ];
+        if (!$markdown) {
+            $element['emoji'] = $emoji;
+            $element['verbatim'] = $verbatim;
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php:36]: From the documentation: This field is only usable when type is mrkdwn.

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $this->options['type'] = 'context';
+    }
+
+    /**
+     * @return $this
+     */
+    public function text(string $text, bool $markdown = true, bool $emoji = true, bool $verbatim = false): self
+    {
+        if (10 === \count($this->options['elements'] ?? [])) {
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php]: given that both images and texts count for the number of elements, I'd use a constant for the limit.

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+final class SlackContextBlock extends AbstractSlackBlock
+{
+    public function __construct()
+    {
+        $this->options['type'] = 'context';
+    }
+
+    /**
+     * @return $this
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php]: return type hint is self-explanatory

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            $element['verbatim'] = $verbatim;
+        }
+        $this->options['elements'][] = $element;
+
+        return $this;
+    }
+
+    /**
+     * @return $this
+     */
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php]: same

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        $this->options['elements'][] = $element;
+
+        return $this;
+    }
+
+    public function image(string $url, string $text): self
+    {
+        if (self::ELEMENT_LIMIT === \count($this->options['elements'] ?? [])) {
+            throw new \LogicException('Maximum number of elements should not exceed 10.');
```
Comment (jderusse) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php]: reuse constant here. same above

**Review by jderusse — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
```
Comment (fabpot) [inline src/Symfony/Component/Notifier/Bridge/Slack/Block/SlackContextBlock.php:12]: Should be removed as we don't use it in Symfony.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @norkunas.

**Pull request merged**